src/shogun/mathematics/linalgrefactor/BaseVector.h

```diff
@@ -0,0 +1,20 @@
+#include <shogun/lib/config.h>
+#include <shogun/lib/SGVector.h>
+
+#ifndef _BASEVECTOR_H__
```

Comment:
get rid of the leading '_'. those are reserved for system headers. We should just use `BASEVECTOR_H__`